### PR TITLE
feat(storage): introduce merge group metrics and corrects some config

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
1. introduce merge config metrics
2. remove unused config `cut_table_size_mb`
3. correct the config name to `periodic_scheduling_compaction_group_interval_sec`

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/meta/src/rpc/metrics.rs [hunk 1]
```diff
@@ -729,6 +732,14 @@ impl MetaMetrics {
         let compaction_event_loop_iteration_latency =
             register_histogram_with_registry!(opts, registry).unwrap();
 
+        let merge_compaction_group_count = register_int_counter_vec_with_registry!(
+            "storage_merge_compaction_group_count",
+            "Count of trigger merge compaction group",
+            &["group_left", "group_right"],
```
Comment (hzxa21) [inline src/meta/src/rpc/metrics.rs:738]: There will be only one data point per `(group_left, group_right)` label pairs for this metric, which means we will have a lot of inactive timeseries most of the time. We should either have a way to safely remove metrics label via `remove_label_values` or emit the metric with fewer labels.

Comment (Li0k) [reply src/meta/src/rpc/metrics.rs:738]: Indeed, I missed the point that the metrics are sufficient to record only the count. There are two points of view
1. merges don't happen very often, and the metrics are more about counts than groups_info. 
2. only `group_left` label is reported, reducing the number of labels

what do you think  @hzxa21?

Comment (hzxa21) [reply src/meta/src/rpc/metrics.rs]: SGTM

**Review by hzxa21 — APPROVED** [review]
LGTM

**Pull request merged**